accessible/fieldview.component.js

```diff
@@ -0,0 +1,123 @@
+        <li #option *ngFor='#optionValue of getOptions(field)' id='{{treeService.createId(option)}}' role='treeitem' aria-selected=false [attr.aria-level]='level+1'>
+          <button #optionButton id='{{treeService.createId(optionButton)}}' (click)='handleDropdownChange(field,optionValue)'>{{optionText[optionValue]}} button</button>
+        </li>
+      </ol>
+      {{setLabelledBy(listItem, concatStringWithSpaces('blockly-argument-menu', label.id))}}
+    </li>
+    <li #listItem aria-selected=false role='treeitem' id='{{treeService.createId(listItem)}}' [attr.aria-level]='level' *ngIf='isCheckbox(field)'>
+      //TODO(madeeha):CHECKBOX
+    </li>
+    <li #listItem aria-selected=false role='treeitem' id='{{treeService.createId(listItem)}}' [attr.aria-level]='level' *ngIf='isTextField(field) && notWhitespace(field)'>
+      <label #label id='{{treeService.createId(label)}}'>
+        {{field.getText()}}
+      </label>
+      {{setLabelledBy(listItem, concatStringWithSpaces('blockly-argument-text', label.id))}}
+    </li>
+    `,
+    inputs: ['field', 'level', 'index', 'parentId'],
+  })
+  .Class({
+    constructor: [blocklyApp.TreeService, function(_service) {
+      this.optionText = {
+        keys: []
+      };
+      this.text = 'Nothing';
+      this.treeService = _service;
+    }],
+    setLabelledBy: function(item,string){
+      if (!item.getAttribute('aria-labelledby')) {
+        item.setAttribute('aria-labelledby', string);
+      }
+    },
+    concatStringWithSpaces: function(a,b){
+      return a + ' ' + b;
+    },
+    isTextInput: function(field) {
+      return field instanceof Blockly.FieldTextInput;
+    },
+    isDropdown: function(field) {
+      return field instanceof Blockly.FieldDropdown;
+    },
+    isCheckbox: function(field) {
+      return field instanceof Blockly.FieldCheckbox;
+    },
+    isTextField: function(field) {
+      return !(field instanceof Blockly.FieldTextInput) &&
+          !(field instanceof Blockly.FieldDropdown) &&
+          !(field instanceof Blockly.FieldCheckbox);
+    },
+    notWhitespace: function(field) {
+      var text = field.getText().trim();
+      return !!text;
+    },
+    getOptions: function(field) {
+      this.optionText.keys.length = 0;
+      var options = field.getOptions_();
+      for (var i = 0; i < options.length; i++) {
+        var tuple = options[i];
+        this.optionText[tuple[1]] = tuple[0];
+        this.optionText.keys.push(tuple[1]);
+      }
+      return this.optionText.keys;
+    },
+    isSelected: function(field, value) {
+      if (value == field.getValue()) {
+        //true will result in the 'selected' option being ENABLED
+        return 'true';
+      }
+      //undefined will result in the 'selected' option being DISABLED
+    },
+    handleDropdownChange: function(field, text) {
+      if (text == 'NO_ACTION') {
+        return;
+      }
+      if (field instanceof Blockly.FieldVariable) {
+        console.log(field);
```

Comment:
drop this, or add some additional text to explain what is being logged here, so that it isn't lost in a sea of console logs of various objects.